### PR TITLE
Adding warning in uninformative graphs.

### PR DESCRIPTION
**Pull request by dharmangbhavsar**

### CODE REVIEW

**Comment by xarkes** [comment]
Hi,
A good practice is to compile your code on your computer, and make sure it works before sending a PR.
As you can see, you have a typo line 155.
More comments below.

File: src/widgets/DisassemblerGraphView.cpp [hunk 1]
```diff
@@ -152,6 +152,13 @@ void DisassemblerGraphView::loadCurrentGraph()
     .set("asm.lines", false)
     .set("asm.lines.fcn", false);
     QJsonDocument functionsDoc = Core()->cmdj("agJ " + RAddressString(seekable->getOffset()));
+    if(fundtionsDoc.isempty())
```
Comment (xarkes) [inline src/widgets/DisassemblerGraphView.cpp]: TYPO

File: src/widgets/DisassemblerGraphView.cpp [hunk 2]
```diff
@@ -152,6 +152,13 @@ void DisassemblerGraphView::loadCurrentGraph()
...
+    {
+        QWidget *warningwidget = new QWidget(this);
```
Comment (xarkes) [inline src/widgets/DisassemblerGraphView.cpp]: I don't think this is needed, you can set the QLabel parent to the current widget (this). Remove that line.

File: src/widgets/DisassemblerGraphView.cpp [hunk 3]
```diff
@@ -152,6 +152,13 @@ void DisassemblerGraphView::loadCurrentGraph()
...
+        QLabel *label = new QLabel(*warningwidget);
```
Comment (xarkes) [inline src/widgets/DisassemblerGraphView.cpp]: new QLabel(this)

File: src/widgets/DisassemblerGraphView.cpp [hunk 4]
```diff
@@ -152,6 +152,13 @@ void DisassemblerGraphView::loadCurrentGraph()
     .set("asm.lines", false)
     .set("asm.lines.fcn", false);
     QJsonDocument functionsDoc = Core()->cmdj("agJ " + RAddressString(seekable->getOffset()));
+    if(fundtionsDoc.isempty())
+    {
```
Comment (xarkes) [inline src/widgets/DisassemblerGraphView.cpp]: Should be on the same line as the if

File: src/widgets/DisassemblerGraphView.cpp (hunk 1 above)
Comment (xarkes) [reply src/widgets/DisassemblerGraphView.cpp]: Missing space before `('

**Comment by xarkes** [comment]
Now I suggest you to fix this, compile it on your side, and if it works, push again :)
Thanks for the contribution!

**Comment by xarkes** [comment]
ping @dharmangbhavsar 
Maybe will you have some time this weekend to fix it?

**Comment by dharmangbhavsar** [comment]
Oh sorry @xarkes . I was trying to build radare2 from source and got into some dependency issues. That is the reason I wasn't able to compile and wanted to complete this fast. Really, really sorry. And sorry again, I have been travelling since a day or two. Will definitely complete this today itself. Sorry for the inconvenience.

**Comment by dharmangbhavsar** [comment]
Hi @xarkes . I tried debugging, but the cmake build for OSX seems to be failing. Can you help with that?
Thank you so much.

**Comment by xarkes** [comment]
```

[32mInstalling ccache[0m



No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```
I think it's an issue on Travis side.

File: src/widgets/DisassemblerGraphView.cpp [hunk 5]
```diff
@@ -152,6 +153,11 @@ void DisassemblerGraphView::loadCurrentGraph()
     .set("asm.lines", false)
     .set("asm.lines.fcn", false);
     QJsonDocument functionsDoc = Core()->cmdj("agJ " + RAddressString(seekable->getOffset()));
+    if (functionsDoc.isEmpty()) {
+        QLabel *label = new QLabel(this);
```
[the remark this answers is not in the document]
Comment (dharmangbhavsar) [reply src/widgets/DisassemblerGraphView.cpp]: Okay. So you want the Qlabel object to be destroyed if a valid graph needs to be displayed?

[the remark this answers is not in the document]
Comment (xarkes) [reply src/widgets/DisassemblerGraphView.cpp]: Not necessarily, but maybe it needs to be hidden, no?
Can you post a screenshot here? When a graph is empty, and after the graph is filled?

**Comment by xarkes** [comment]
ping @dharmangbhavsar

**Comment by xarkes** [comment]
Closed in favor of https://github.com/radareorg/cutter/commit/22465615a4cda2c9bf6a2efd5667b62dba1d8814

**Pull request closed without merging**